inbound/naive.go

```diff
@@ -420,29 +251,34 @@ func (c *naiveH2Conn) read(p []byte) (n int, err error) {
 		c.paddingRemaining = 0
 	}
 	if c.readPadding < kFirstPaddings {
-		paddingHdr := p[:3]
-		_, err = io.ReadFull(c.reader, paddingHdr)
-		if err != nil {
-			return
-		}
-		originalDataSize := int(binary.BigEndian.Uint16(paddingHdr[:2]))
-		paddingSize := int(paddingHdr[2])
-		if len(p) > originalDataSize {
-			p = p[:originalDataSize]
+		c.readPadding++
+		nr, err := io.ReadFull(c.reader, p[0:3])
+		if nr > 0 {
+			nr = int(p[0])*256 + int(p[1])
```

Comment:
This is exactly BigEndian uint16, don't understand or do you prefer unreadable writing?